src/Data/ProgrammesDb/EntityRepository/PromotionRepository.php

```diff
@@ -23,38 +27,19 @@ public function findActivePromotionsByContext(int $contextId, DateTimeImmutable
             ->andWhere('promotion.isActive = 1')
             ->andWhere('promotion.startDate <= :datetime')
             ->andWhere('promotion.endDate > :datetime')
-            ->andWhere('promotion.context = :contextId')
             ->addOrderBy('promotion.weighting', 'ASC')
             ->setFirstResult($offset)
             ->setMaxResults($limit)
-            ->setParameter('contextId', $contextId)
             ->setParameter('datetime', $datetime);
 
-        $promotions = $qb->getQuery()->getResult(Query::HYDRATE_ARRAY);
-        return $this->resolveParentsForPromosOfCoreEntities($promotions);
-    }
-
-    /**
-     * @param int[] $ancestryIds
-     * @return array[]
-     */
-    public function findActiveSuperPromotionsByAncestry(array $ancestryIds, DateTimeImmutable $datetime, ?int $limit, int $offset): array
-    {
-        $qb = $this->createQueryBuilder('promotion')
-           ->addSelect(['promotionOfCoreEntity', 'promotionOfImage', 'imageForPromotionOfCoreEntity'])
-           ->leftJoin('promotion.promotionOfCoreEntity', 'promotionOfCoreEntity')
-           ->leftJoin('promotion.promotionOfImage', 'promotionOfImage')
-           ->leftJoin('promotionOfCoreEntity.image', 'imageForPromotionOfCoreEntity')
-           ->andWhere('promotion.isActive = 1')
-           ->andWhere('promotion.startDate <= :datetime')
-           ->andWhere('promotion.endDate > :datetime')
-           ->andWhere('promotion.context in (:ancestryIds)')
-           ->andWhere('promotion.cascadesToDescendants = 1')
-           ->addOrderBy('promotion.weighting', 'ASC')
-           ->setFirstResult($offset)
-           ->setMaxResults($limit)
-           ->setParameter('ancestryIds', $ancestryIds)
-           ->setParameter('datetime', $datetime);
+        if (!$ancestryIds) {
```

Comment:
Good call, an explanatory comment on what this is doing and why would be handy too.